src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.iOSmacOS.cs

```diff
@@ -0,0 +1,1696 @@
+			// returning a value from our cache prevents glitchy animations.
+			var isUpdatingItem = _updateItems?.Any(item => item.IndexPathAfterUpdate?.Equals(itemIndexPath) ?? false) ?? false;
+			if (!isUpdatingItem)
+			{
+				var unoCachedAttributes = LayoutAttributesForItem(itemIndexPath);
+				if (unoCachedAttributes != null)
+				{
+					attributes.Frame = unoCachedAttributes.Frame;
+				}
+			}
+
+			return attributes;
+		}
+#if __IOS__
+		public override _LayoutAttributes FinalLayoutAttributesForDisappearingItem(NSIndexPath itemIndexPath)
+		{
+			var attributes = base.FinalLayoutAttributesForDisappearingItem(itemIndexPath);
+#else
+		public override _LayoutAttributes GetFinalLayoutAttributesForDisappearingItem(NSIndexPath itemIndexPath)
+		{
+			var attributes = base.GetFinalLayoutAttributesForDisappearingItem(itemIndexPath);
+#endif
+			//For some reason (dynamic item sizing?) this can be called not only for disappearing items but any visible item. In this case 
+			// returning a value from our cache prevents glitchy animations.
+			var isUpdatingItem = _updateItems?.Any(item => item.IndexPathBeforeUpdate?.Equals(itemIndexPath) ?? false) ?? false;
+			if (!isUpdatingItem)
+			{
+				var unoCachedAttributes = LayoutAttributesForItem(itemIndexPath);
+				if (unoCachedAttributes != null && attributes != null)
+				{
+					attributes.Frame = unoCachedAttributes.Frame;
+				}
+			}
+
+			return attributes;
+		}
+#endregion
+
+		public CGSize SizeThatFits(CGSize size)
+		{
+			TrySetHasConsumedUnusedSpace();
+			return PrepareLayout(false, size);
+		}
+
+		/// <summary>
+		/// Prepare layout if necessary and return its size.
+		/// </summary>
+		/// <param name="createLayoutInfo">Should we create <see cref="_LayoutAttributes"/>?</param>
+		/// <param name="size">The available size</param>
+		/// <returns>The total collection size</returns>
+		/// <remarks>This is called by overridden methods which need to know the total dimensions of the panel content. If a full relayout is required,
+		/// it calls <see cref="PrepareLayoutInternal(bool, bool, CGSize)"/>; otherwise it returns a cached value.</remarks>
+		private CGSize PrepareLayout(bool createLayoutInfo, CGSize? size = null)
+		{
+			using (
+			   _trace.WriteEventActivity(
+				   TraceProvider.ListViewBaseLayout_PrepareLayoutStart,
+				   TraceProvider.ListViewBaseLayout_PrepareLayoutStop
+			   )
+			)
+			{
+				if (size.HasValue && _lastAvailableSize != size)
+				{
+					if (IsLayoutRequiredOnSizeChange(_lastAvailableSize, size.Value))
+					{
+						_dirtyState = DirtyState.NeedsRelayout;
+					}
+
+					if (!size.Value.IsEmpty)
+					{
+						_lastAvailableSize = size.Value;
+					}
+				}
+
+				if (_dirtyState == DirtyState.NeedsRelayout || _dirtyState == DirtyState.CollectionChanged)
+				{
+					var availableSize = size ?? AvailableSize;
+					if (createLayoutInfo)
+					{
+						_lastElement = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'groupHeaderLayout' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4931087)